lib/graphql/input_object_type.rb

```diff
@@ -51,6 +51,18 @@ def kind
     def validate_non_null_input(input, warden)
       result = GraphQL::Query::InputValidationResult.new
 
+      if input.is_a? Array
```

Comment:
Did a check for Array and Enumerable instead of Hash, because you use `MinimumInputObject` (spec/support/minimum_input_object.rb) in your specs. So I'm not sure if input could be something else than a Hash. I could rewrite MinimumInputObject to be a Hash or leave it like this. Your call!